deep_qa/training/text_trainer.py

```diff
@@ -491,10 +472,12 @@ def _set_padding_lengths_from_model(self):
 
     ########################
     # Model-specific methods - if you do anything complicated, you probably need to override these,
-    # but simple models might be able to get by with just the default implementation
+    # but simple models might be able to get by with just the default implementation.  Some of
```

Comment:
Maybe we should just make all of these methods not private as they are used in subclasses (I know this is not the standard definition of private) and make all the ones which are currently `__func` private. The `__func` does weird class name substitution things and I don't like it - would you agree that this usage is pretty unusual to find in Python code in general?